linera-views/src/scylla_db.rs

```diff
@@ -861,103 +749,70 @@ impl WritableKeyValueStore<ScyllaDbContextError> for ScyllaDbStore {
     }
 }
 
+#[async_trait]
+impl AdminKeyValueStore<ScyllaDbContextError> for ScyllaDbStore {
+    type Config = ScyllaDbStoreConfig;
+
+    async fn connect(config: &Self::Config, namespace: &str) -> Result<Self, ScyllaDbContextError> {
+        let cache_size = config.common_config.cache_size;
+        let simple_store = ScyllaDbStoreInternal::connect(config, namespace).await?;
+        let store = Self::get_complete_store(simple_store, cache_size);
+        Ok(store)
+    }
+
+    async fn list_all(config: &Self::Config) -> Result<Vec<String>, ScyllaDbContextError> {
+        ScyllaDbStoreInternal::list_all(config).await
+    }
+
+    async fn delete_all(config: &Self::Config) -> Result<(), ScyllaDbContextError> {
+        ScyllaDbStoreInternal::delete_all(config).await
+    }
+
+    async fn exists(config: &Self::Config, namespace: &str) -> Result<bool, ScyllaDbContextError> {
+        ScyllaDbStoreInternal::exists(config, namespace).await
+    }
+
+    async fn create(config: &Self::Config, namespace: &str) -> Result<(), ScyllaDbContextError> {
+        ScyllaDbStoreInternal::create(config, namespace).await
+    }
+
+    async fn delete(config: &Self::Config, namespace: &str) -> Result<(), ScyllaDbContextError> {
+        ScyllaDbStoreInternal::delete(config, namespace).await
+    }
+}
+
 impl KeyValueStore for ScyllaDbStore {
     type Error = ScyllaDbContextError;
 }
 
 impl ScyllaDbStore {
     /// Gets the table name of the ScyllaDB store.
     #[cfg(not(feature = "metrics"))]
-    pub async fn get_table_name(&self) -> String {
-        self.store.store.store.get_table_name().await
+    pub async fn get_namespace(&self) -> String {
+        self.store.store.store.get_namespace().await
     }
 
     /// Gets the table name of the ScyllaDB store.
     #[cfg(feature = "metrics")]
-    pub async fn get_table_name(&self) -> String {
-        self.store.store.store.store.store.get_table_name().await
+    pub async fn get_namespace(&self) -> String {
+        self.store.store.store.store.store.get_namespace().await
     }
 
     #[cfg(not(feature = "metrics"))]
-    fn get_complete_store(
-        store: JournalingKeyValueStore<ScyllaDbStoreInternal>,
-        cache_size: usize,
-    ) -> Self {
+    fn get_complete_store(simple_store: ScyllaDbStoreInternal, cache_size: usize) -> Self {
+        let store = JournalingKeyValueStore::new(simple_store);
         let store = LruCachingStore::new(store, cache_size);
         Self { store }
     }
 
     #[cfg(feature = "metrics")]
-    fn get_complete_store(
-        store: JournalingKeyValueStore<ScyllaDbStoreInternal>,
-        cache_size: usize,
-    ) -> Self {
+    fn get_complete_store(simple_store: ScyllaDbStoreInternal, cache_size: usize) -> Self {
+        let store = JournalingKeyValueStore::new(simple_store);
         let store = MeteredStore::new(&SCYLLA_DB_METRICS, store);
```

Comment:
Should we just put the `#[cfg(feature = "metrics")]` on this line and 813?

Reply:
I agree. This simplifies the code. And this was done for RocksDB and DynamoDB.